core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -379,8 +379,8 @@ public CheckpointName getCheckpointName() {
 
   @Override
   public CloseableIterator<JournalEntry> getJournalEntryIterator() {
-    Iterator<Block> blockStoreIterator = mBlockStore.iterator();
-    Iterator<JournalEntry> blockIterator = new Iterator<JournalEntry>() {
+    CloseableIterator<Block> blockStoreIterator = mBlockStore.getCloseableIterator();
+    Iterator<JournalEntry> journalIterator = new Iterator<JournalEntry>() {
```

Comment:
If this Iterator translation looks so annoying, I will need to change the `CloseableIterator` API a bit. Now `class CloseableIterator<T> extends CloseableResource<Iterator<T>> implements Iterator<T>` so the `CloseableIterator<JournalEntry>` can only be constructed with `Iterator<JournalEntry>`, this translate must happen outside CloseableIterator. 

The CloseableIterator can be changed to CloseableIterator<A, B> to take Iterator<A> and returns CloseableIterator<B>. This includes a bit more work on the CloseableIterator and here is currently the only use case. So I'm not adding it unless reviewers think having that will make the code much cleaner.

Reply:
From the current scope maybe adding a TODO for myself makes more sense.